src/audio/dai.c

```diff
@@ -705,10 +705,10 @@ static int dai_config(struct comp_dev *dev, struct sof_ipc_dai_config *config)
 		return -EINVAL;
 	}
 
-	switch (config->type) {
+	switch (config->info.type) {
 	case SOF_DAI_INTEL_SSP:
 		/* set dma burst elems to slot number */
-		dd->config.burst_elems = config->ssp.tdm_slots;
+		dd->config.burst_elems = config->ssp[0].tdm_slots;
```

Comment:
rather than using 0, we should use a struct member (hard coded to 0 today), that way the final functional update becomes easier.

Reply:
Not sure what you mean.
Are you talking about simply dereferencing 'ssp' instead of indexing it with 0?

Reply:
@lgirdwood sorry, mind giving me a hint here?

Reply:
Sorry, I mean
```
dd->dai_index = 0;
...
dd->config.burst_elems = config->ssp[dd->dai_index].tdm_slots;
```

Reply:
Ah, in the DAI data. Got it, cheers

Reply:
Hmm, on the other hand, this creates an inconsistency, because the drivers will still resort to indexing the 0.

One way I'd see here is to have the proposed dai_index (or rather config_index) inside of the DAI config IPC. However, this creates a potential issue if the data is not initialized to 0 in the IPC.

Reply:
@slawblauciak doesn't matter where you store the index today, as long as we have an index rather than hard coding to 0. The DAI config switching logic will no doubt place the index in the correct [place when it's implemented.